src/goalServices/goals.js

```diff
@@ -2637,8 +2637,18 @@ export async function mergeObjectiveFromGoal(objective, parentGoalId) {
   objective.activityReportObjectives.forEach((aro) => {
     updatesToRelatedModels.push(
       ActivityReportObjective.update({
-        objectiveId: newObjective.id,
         originalObjectiveId: aro.objectiveId,
+      }, {
+        where: {
+          id: aro.id,
+          originalObjectiveId: null,
+        },
+        individualHooks: true,
+      }),
+    );
+    updatesToRelatedModels.push(
+      ActivityReportObjective.update({
+        objectiveId: newObjective.id,
```

Comment:
We want to do it as a single update for that record, as its not good to try to update the same row twice within the same transaction. Row locking starts getting tricky.

Could we instead do something like sequelize.literal(`COALESCE("originalObjectiveId", ${aro.objectiveId})`),